### PR TITLE
Support Date objects

### PR DESCRIPTION
**Pull request by SferaDev**
- Make client to parse and return "Date" objects for ``datetime`` fields.
- Do not depend on codegen ``links`` for schema retrieval
- Update unit tests to use ``msw``

Future improvement:

- Accept ``Date | string`` on create/update operations https://github.com/xataio/client-ts/issues/191

### CODE REVIEW

File: test/integration.test.ts [hunk 1]
```diff
@@ -775,21 +775,21 @@ describe('getBranch', () => {
 
     const getBranchOptions = { apiKey: '', apiUrl: '', fetchImpl: {} as FetchImpl };
 
-    process.env = { XATA_BRANCH: branchName };
+    process.env = { NODE_ENV: 'development', XATA_BRANCH: branchName };
```
Comment (gimenete) [inline test/integration.test.ts:778]: Just curious. Why do we need this env now?

File: test/integration/dates.test.ts [hunk 2]
```diff
@@ -73,7 +73,7 @@ describe('dates', () => {
     const date = new Date();
     const record = await xata.db.datetime.create({ date });
 
-    // TODO: Parse date in the client
-    expect(record.date).toEqual(date.toISOString());
+    expect(typeof record.date).toEqual('object');
```
Comment (gimenete) [inline test/integration/dates.test.ts:76]: What do you think?

```suggestion
    expect(record.date instanceof Date).toEqual(true);
```

**Review by gimenete — APPROVED** [review]
LGTM! One question and one suggestion.

File: test/integration.test.ts (hunk 1 above)
Comment (SferaDev) [reply test/integration.test.ts:778]: We don't for the tests to pass, but ``tsc`` complains that ``NodeEnv`` always needs to be set. Adding it to fix type errors from the compiler

**Pull request merged**